How to configure proxy

I'm trying to use it on a server that needs proxy.
I've tried setting TEAMCITY_SERVER_OPTS with -Dhttp.proxyHost=.... -Dhttp.proxyPort=... -Dhttps.proxyHost=.... -Dhttps.proxyPort=.... -Dhttp.nonProxyHosts=.... 
and restarting teamcity, then even uninstall and install the plugin again.

Also tried to add

teamcity.http.nonProxyHosts=...
teamcity.https.proxyHost=...
teamcity.https.nonProxyHosts=...
teamcity.http.proxyHost=...
teamcity.https.proxyPort=...
teamcity.http.proxyPort=...

on internal.properties

but everytime I get this:

`[2021-02-13 23:42:24,271]  ERROR [9@58f69f86'; Normal executor 1] - notificator.DiscordNotificator - Failed to send the WebHook!
org.apache.http.conn.HttpHostConnectException: Connect to discord.com:443 [discord.com/162.159.136.232, discord.com/162.159.135.232, discord.com/162.159.128.233, discord.com/162.159.137.232, discord.com/162.159.138.232] failed: Connection timed out (Connection timed out)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at com.github.playerforcehd.tcdiscordwebhooks.discord.DiscordWebHookProcessor.sendDiscordWebHook(DiscordWebHookProcessor.java:95)
	at com.github.playerforcehd.tcdiscordwebhooks.discord.DiscordWebHookProcessor.sendDiscordWebHook(DiscordWebHookProcessor.java:73)
	at com.github.playerforcehd.tcdiscordwebhooks.notificator.DiscordNotificator.processNotify(DiscordNotificator.java:145)
	at com.github.playerforcehd.tcdiscordwebhooks.notificator.DiscordNotificator.notifyBuildFailed(DiscordNotificator.java:258)
	at jetbrains.buildServer.notification.impl.UserRulesAwareNotificationListener$NotificatorEventAdapter.notifyBuildFailureWatchers(UserRulesAwareNotificationListener.java:237)
	at jetbrains.buildServer.notification.impl.UserRulesAwareNotificationListener$NotificatorEventAdapter.buildFailedSubsequent(UserRulesAwareNotificationListener.java:192)
	at jetbrains.buildServer.notification.impl.UserRulesAwareNotificationListener$NotificatorEventAdapter.access$400(UserRulesAwareNotificationListener.java:148)
	at jetbrains.buildServer.notification.impl.UserRulesAwareNotificationListener.buildFailed(UserRulesAwareNotificationListener.java:23)
	at jetbrains.buildServer.notification.NotificationProcessor$5.call(NotificationProcessor.java:3)
	at jetbrains.buildServer.notification.NotificationProcessor$19.run(NotificationProcessor.java:8)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.net.ConnectException: Connection timed out (Connection timed out)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 22 more
`

--- a/tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/discord/DiscordWebHookProcessor.java
+++ b/tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/discord/DiscordWebHookProcessor.java
@@ -89,11 +89,11 @@ public class DiscordWebHookProcessor {
             String httpProxyHost = System.getProperty("http.proxyHost");
             String httpProxyPort = System.getProperty("http.proxyPort");
             if (httpProxyHost != null && httpProxyPort.trim().length() > 0 && httpProxyPort != null) {
-                Integer port = Integer.valueOf(httpProxyPort);
+                int port = Integer.parseInt(httpProxyPort);
                 HttpHost proxy = new HttpHost(httpProxyHost, port, "http");
-                RequestConfig.Builder reqconfigconbuilder = RequestConfig.custom();
-                reqconfigconbuilder = reqconfigconbuilder.setProxy(proxy);
-                RequestConfig config = reqconfigconbuilder.build();
+                RequestConfig.Builder reqConfigBuilder = RequestConfig.custom();
+                reqConfigBuilder = reqConfigBuilder.setProxy(proxy);
+                RequestConfig config = reqConfigBuilder.build();
                 httpPost.setConfig(config);
             }
             try (CloseableHttpResponse response = httpClient.execute(httpPost)) {

--- a/tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/discord/DiscordWebHookProcessor.java
+++ b/tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/discord/DiscordWebHookProcessor.java
@@ -88,7 +88,7 @@ public class DiscordWebHookProcessor {
             httpPost.setEntity(new StringEntity(discordWebHookPayload, HTTP_CHARSET));
             String httpProxyHost = System.getProperty("http.proxyHost");
             String httpProxyPort = System.getProperty("http.proxyPort");
-            if (httpProxyHost!=null && httpProxyPort.trim().length()>0 && httpProxyPort!=null) {
+            if (httpProxyHost != null && httpProxyPort.trim().length() > 0 && httpProxyPort != null) {
                 Integer port = Integer.valueOf(httpProxyPort);
                 HttpHost proxy = new HttpHost(httpProxyHost, port, "http");
                 RequestConfig.Builder reqconfigconbuilder = RequestConfig.custom();

--- a/tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/discord/DiscordWebHookProcessor.java
+++ b/tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/discord/DiscordWebHookProcessor.java
@@ -17,6 +17,8 @@
 package com.github.playerforcehd.tcdiscordwebhooks.discord;
 
 import com.google.gson.Gson;
+import org.apache.http.HttpHost;
+import org.apache.http.client.config.RequestConfig;
 import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.entity.StringEntity;
@@ -84,6 +86,16 @@ public class DiscordWebHookProcessor {
             httpPost.addHeader("Accept-Language", "en-US,en;q=0.5");
             httpPost.addHeader("Content-Type", "application/json");
             httpPost.setEntity(new StringEntity(discordWebHookPayload, HTTP_CHARSET));
+            String httpProxyHost = System.getProperty("http.proxyHost");
+            String httpProxyPort = System.getProperty("http.proxyPort");
+            if (httpProxyHost!=null && httpProxyPort.trim().length()>0 && httpProxyPort!=null) {
+                Integer port = Integer.valueOf(httpProxyPort);
+                HttpHost proxy = new HttpHost(httpProxyHost, port, "http");
+                RequestConfig.Builder reqconfigconbuilder = RequestConfig.custom();
+                reqconfigconbuilder = reqconfigconbuilder.setProxy(proxy);
+                RequestConfig config = reqconfigconbuilder.build();
+                httpPost.setConfig(config);
+            }
             try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
                 responseCode = response.getStatusLine().getStatusCode();
             }

